prov/efa/src/rxr/rxr_read.c

```diff
@@ -79,15 +79,46 @@ int rxr_locate_rma_iov_pos(struct fi_rma_iov *rma_iov, int rma_iov_count, size_t
 	return -1;
 }
 
+ssize_t rxr_read_mr_reg(struct rxr_ep *ep, struct rxr_read_entry *read_entry)
+{
+	size_t i;
+	int err;
+
+	for (i = 0; i < read_entry->iov_count; ++i) {
+		if (read_entry->mr_desc[i] || read_entry->mr[i]) {
+			continue;
+		}
+
+		err = fi_mr_reg(rxr_ep_domain(ep)->rdm_domain,
+				read_entry->iov[i].iov_base, read_entry->iov[i].iov_len,
+				FI_RECV, 0, 0, 0, &read_entry->mr[i], NULL);
+
+		if (err) {
+			/* In case error happened, we did not close registered MR here, because:
```

Comment:
sorry, the comment is a little misleading. What I meant to say is: in case of multiple iov, if the i-th registration failed, ```mr[0]``` to ```mr[i-1]``` are still registered, and need to be closed eventually. We do not close them here, because they will be closed in ```rxr_read_release_entry```.

Reply:
Thank you, this makes sense to me now.